proto/cosmos/app/v1alpha1/config.proto

```diff
@@ -33,4 +38,18 @@ message ModuleConfig {
   // config is the config object for the module. Module config messages should
   // define a ModuleDescriptor using the cosmos.app.v1alpha1.is_module extension.
   google.protobuf.Any config = 2;
+
+  // golang_bindings specifies explicit interface to implementation type bindings which
+  // depinject uses to resolve interface inputs to provider functions.  The scope of this
+  // field's configuration is module specific.
+  repeated GolangBinding golang_bindings = 3;
 }
+
+// GolangBinding is an explicit interface type to implementing type binding for dependency injection.
+message GolangBinding {
+  // interface_type is the interface type which will be bound to a specific implementation type
+  string interface_type = 1;
+
+  // implementation is the implementing type which will be supplied when an input of type interface is requested
+  string implementation = 2;
```

Comment:
How about simply:
```suggestion
  string iface = 1;

  // implementation is the implementing type which will be supplied when an input of type interface is requested
  string impl = 2;
```

Reply:
Is that too terse?

Reply:
could also be `interface` and `impl`

Reply:
These aren't super long field names. I personally don't see much value add in the abbreviations.